pulp_2to3_migration/app/pre_migration.py

```diff
@@ -397,15 +410,28 @@ def pre_migrate_importer(repo_id, importers, importer_types, repo=None):
     last_updated = (importer_data.last_updated
                     and timezone.make_aware(importer_data.last_updated, timezone.utc))
 
-    Pulp2Importer.objects.create(
+    importer, created = Pulp2Importer.objects.get_or_create(
         pulp2_object_id=importer_data.id,
-        pulp2_type_id=importer_data.importer_type_id,
-        pulp2_last_updated=last_updated,
-        pulp2_config=importer_data.config,
-        pulp2_repository=repo,
-        pulp2_repo_id=repo_id,
-        is_migrated=False
-    )
+        defaults={'pulp2_type_id': importer_data.importer_type_id,
+                  'pulp2_last_updated': last_updated,
+                  'pulp2_config': importer_data.config,
+                  'pulp2_repository': repo,
+                  'pulp2_repo_id': repo_id,
+                  'is_migrated': False})
+
+    if not created:
+        # if it was marked as such because it was not present in the migration plan
+        importer.not_in_plan = False
+        # check if there were any changes since last time
+        if last_updated != importer.pulp2_last_updated:
+            # remove Remote in case of feed change
+            if importer.pulp2_config.get('feed') != importer_data.config.get('feed'):
+                importer.pulp3_remote.delete()
```

Comment:
When we delete a migrated importer, do we need to flip LCEs to is_migrated=False?

Reply:
yes, this is a good catch thank you.